pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/coordination/CoordinationService.java

```diff
@@ -56,4 +56,21 @@ <T> LeaderElection<T> getLeaderElection(Class<T> clazz, String path,
      *             if there's a failure in incrementing the counter
      */
     CompletableFuture<Long> getNextCounterValue(String path);
+
+    /**
+     * Increment a counter identified by the specified path and return the current value.
+     * The counter value will be guaranteed to be unique within the context of the path.
+     * It will retry when {@link org.apache.pulsar.metadata.api.MetadataStoreException} happened.
+     *
+     * If the maximum number of retries is reached and still failed,
+     * the feature will complete with exception {@link org.apache.pulsar.metadata.api.MetadataStoreException}.
+     *
+     * @param path
+     *            The path that identifies a particular counter
+     * @param count
+     *            The retry count.
+     * @return
+     *            A future that will track the completion of the operation
+     */
+    CompletableFuture<Long> getNextCounterValueWithRetry(String path, int count);
```

Comment:
I wouldn't add a new method in the API, rather just doing that internally in the existing API method.

Reply:
@merlimat Do you suggest using a fixed retry count in `CoordinationServiceImpl#getNextCounterValue` method, if I understand correctly?

Reply:
👍 Yes, we could rename: 
 * `getNextCounterValueWithRetry` -> `getNextCounterValue`
 * `getNextCounterValue` -> `internalGetNextCounterValue`

Reply:
Thanks! Please help take a look again.